isamples_metadata/GEOMETransformer.py

```diff
@@ -0,0 +1,362 @@
+import typing
+
+from isamples_metadata.Transformer import (
+    Transformer,
+)
+
+
+class GEOMETransformer(Transformer):
+    """Concrete transformer class for going from a GEOME record to an iSamples record"""
+
+    def transform(self) -> typing.Dict:
+        transformed_record = super(GEOMETransformer, self).transform()
+        return transformed_record
+
+    ARK_PREFIX = "ark:/"
+
+    def _source_record_main_record(self):
+        # The sub-record for the main record, as opposed to the parent or children records
+        return self.source_record["record"]
+
+    def _source_record_parent_record(self):
+        # The sub-record for the parent record
+        return self.source_record["parent"]
+
+    def _id_minus_prefix(self):
+        return self._source_record_main_record()["bcid"].removeprefix(self.ARK_PREFIX)
+
+    def id_string(self):
+        return f"metadata/{self._id_minus_prefix()}"
+
+    def sample_label(self) -> typing.AnyStr:
+        main_record = self._source_record_main_record()
+        label_components = []
+        scientific_name = main_record.get("scientificName")
+        if scientific_name is not None:
+            label_components.append(scientific_name)
+        sample_id = main_record.get("materialSampleID")
+        if sample_id is not None:
+            label_components.append(sample_id)
+        return " ".join(label_components)
+
+    def sample_identifier_string(self) -> typing.AnyStr:
+        return self._source_record_main_record()["bcid"]
+
+    def sample_identifier_scheme(self) -> typing.AnyStr:
+        return "ark"
+
+    def sample_identifier_value(self) -> typing.AnyStr:
+        return "/"
+
+    def sample_description(self) -> typing.AnyStr:
+        # TODO: implement
+        # [concatenate key:value pairs from other content in record/ that is not mapped, e.g.
+        # 'voucherCatalogNumber:USNM:Fish:433156'; record/preservative, ]
```

Comment:
This one was a bit undefined and the examples didn't have data populated, so I skipped it for now.

Reply:
I think after all possible record properties are mapped to iSamples schema, the rest of the record properties are put in the description. This is from smr note. ”lots of stuff gets concatenated here, there is no required description. Some fields that would get stuffed here: chronometric age range, named era age range, basisOfRecord, dynamicProperties, establishmentMeans, genbankSpecimenVoucher, identificationRemarks, identificationVerificationStatus, individualCount, morphospeciesDescription, nomenclaturalCode, occurrenceRemarks, organismID, organismQuantity, organismQuantityType, organismRemarks, otherCatalogNumbers, previousIdentifications, taxonRemarks, typeStatus, verbatimLifeStage, weight, weightUnits, dayIdentified, monthIdentified, yearIdentified, lengthUnits, length, sex“.

Reply:
Thanks @qgan7125 -- I forgot to grab the rest of these.  Thanks for including the list here, will make it much easier to add them all!